cohort_schedules/C07_classroom.md

```diff
@@ -1,39 +1,126 @@
+| CS Fundamentals         | Career Readiness
+
+## Expected Outcomes
+
+### Ruby
+
+Ruby is a dynamically typed, interpreted, object-oriented language. In addition to serving as a great introduction to programming, Ruby is the foundation of the Ruby on Rails framework, and is used for most of our CS Fundamentals curriculum.
+
+Students should be able to:
+
+* Write complex command line programs in Ruby
+* Use object-oriented programming principals to build scalable, maintainable programs
+* Read and write files, and process large amounts of data
+* Verify code from the outset using test-driven development
+
+### HTML & CSS
+
+HTML and CSS are two of the core technologies on which the web is built, allowing web developers to structure content in a hierarchy and specify how it should be presented to the user. Both are foundational for web develoeprs.
+
+Students should be able to:
+
+* Read and write HTML and CSS
+* Utilize semantic tags to provide clear structure to HTML
+* Manage the layout of a complex web page using CSS
+
+### Ruby on Rails
+
+Ruby on Rails is a robust and comprehensive MVC framework for building web servers. We use Rails to address a wide range of topics, from HTTP, REST and web standards to database operations. This unit also includes a large team project completed over two weeks.
+
+Students should be able to:
+
+* Build and deploy a complete web application, from the HTML & CSS on the page to the backing database
+* Model complex real-world relations using ActiveRecord
+* Use Git, TDD and agile practices to coordinate a large team
+
+### Web APIs
+
+APIs allow programs to talk to each other, and have become an important part of the modern web. We focus both on consuming third party APIs and on producing our own APIs, laying the foundation for future front-end development.
+
+Students should be able to:
+
+* Explore and utilize third party APIs
+* Design an API to expose their own data
+* Write programs that parse and format JSON
+
+### JavaScript
+
+As the language of the web, an understanding of JavaScript is essential for any prospective web developer. In addition to core language features, we'll cover some of the excellent libraries that enable modern web apps such as jQuery, Underscore and the Backbone framework.
+
+Students should be able to:
+
+* Read, write and test JavaScript
+* Manipulate the DOM dynamically
+* Build a complete single-page application, using AJAX to integrate with an API
+* Organize a web app using a front-end framework
+
+### Computer Science Fundamentals
+
+Ada graduates are expected to have an understanding of the basics of computer science. This includes concepts such as how computers and operating systems work, common data structures and algorithms, and notions of space- and time-efficiency.
+
+Students should be able to:
+
+* Implement various sort and search algorithms
+* Select the appropriate data structure to solve a given problem
+* Understand and implement recursive functions
```

Comment:
I do -- but it's very guided. They don't do a ton of the thinking. They just follow the process.